src/dotnetTemplate/Microsoft.PowerShell.PowerShellStandard.Template/Microsoft.PowerShell.PowerShellStandard.Template/InvokeNoun.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Management.Automation;
+using System.Management.Automation.Runspaces;
+
+namespace Microsoft.PowerShell.PowerShellStandard.Template
+{
+    [Cmdlet("Invoke","Noun")]
+    public class InvokeNoun : PSCmdlet
+    {
+        [Parameter(
+            Mandatory=true,
+            Position = 0,
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true)]
+        public string Value { get; set; }
+
+        [Parameter()]
+        [ValidateSet("Output", "Verbose", "Debug")]
+        [Alias(("To"))]
+        public string WriteTo { get; set; } = "Output";
+
+        protected override void BeginProcessing()
+        {
+            WriteObject("Begin!");
```

Comment:
Maybe use WriteWarning instead of showing how to write strings to the pipeline.  This sample should have some OutputType that does get written to the pipeline to show proper cmdlet design.

Reply:
Can you elaborate on this a bit more? I don't have too much experience with binary modules.

Are you looking for the cmdlet to return some object?